tests/impl/async_start_transaction.cpp

```diff
@@ -0,0 +1,113 @@
+#include "connection_mock.h"
+
+#include <ozo/impl/async_start_transaction.h>
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+namespace {
+
+using namespace testing;
+using namespace ozo::tests;
+
+using ozo::error_code;
+
+struct async_start_transaction : Test {
+    StrictMock<connection_gmock> connection {};
+    StrictMock<executor_gmock> callback_executor{};
+    StrictMock<callback_gmock<ozo::impl::transaction<connection_ptr<>>>> callback {};
+    StrictMock<executor_gmock> executor {};
+    StrictMock<executor_gmock> strand {};
+    StrictMock<strand_executor_service_gmock> strand_service {};
+    StrictMock<stream_descriptor_gmock> socket {};
+    StrictMock<steady_timer_gmock> timer {};
+    io_context io {executor, strand_service};
+    decltype(make_connection(connection, io, socket, timer)) conn = make_connection(connection, io, socket, timer);
+};
+
+TEST_F(async_start_transaction, should_call_async_execute) {
+    *conn->handle_ = native_handle::good;
+
+    Sequence s;
+
+    EXPECT_CALL(executor, dispatch(_)).InSequence(s).WillOnce(InvokeArgument<0>());
+    EXPECT_CALL(strand_service, get_executor()).InSequence(s).WillOnce(ReturnRef(strand));
+
+    // Send query params
```

Comment:
Too low level, same as here https://github.com/yandex/ozo/pull/58/files/8c25fe0a548b057b7092cc38d9e9eae16c0b2bbb#diff-b2000bf70b252ad7f99cd703250f1f8dR39

In this tests async_execute() call should be mocked, that is why you need different connection mocks.

Reply:
Done